File: app/src/main/java/com/raeed/tictactoc/MyAdapter.java
package com.raeed.tictactoc;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;
import java.util.ArrayList;

public class MyAdapter extends BaseAdapter {
    HistoryArray information;
    private Context context;
    private ArrayList<HistoryArray> historyArrays = new ArrayList<>();
    private int resorsec;

    public MyAdapter(Context context, int resorsec ,ArrayList<HistoryArray> historyArrays) {
        this.context = context;
        this.historyArrays = historyArrays;
        this.resorsec=resorsec;
    }



    @Override
    public int getCount()
    {
        return historyArrays.size();
    }

    @Override
    public Object getItem(int position) {
        information= historyArrays.get(position);
        return information;
    }

    @Override
    public long getItemId(int position) {

        return 0;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        View view;
        view =convertView;
        if (view==null){
        view= LayoutInflater.from(context).inflate(R.layout.listviewlayout,null,false);}
        ImageView player_A_Image = view.findViewById(R.id.Aimage);
        ImageView player_B_Image = view.findViewById(R.id.Bimage);
        TextView aVsb=view.findViewById(R.id.AvsB);
        TextView serialnumber= view.findViewById(R.id.number);
        TextView player_A_ruselt= view.findViewById(R.id.Atext);
        TextView player_B_ruselt= view.findViewById(R.id.Btext);

        player_A_Image.setImageResource(R.drawable.test);
        player_B_Image.setImageResource(R.drawable.test);

        String player_A=information.getPlayer_A(); // Todo حل مشكلة null object refernce
        String player_B=information.getPlayer_B();

        aVsb.setText(player_A+ " Vs " + player_B);
        serialnumber.setText(position+"");
        int id=information.getWiner_Id();

        if (id==0) {
            player_A_ruselt.setText("winer");
            player_B_ruselt.setText("Looser");
            } else {
            player_A_ruselt.setText("Looser");
            player_B_ruselt.setText("winer");
        }
        return view;
    }
}

File: app/src/main/java/com/raeed/tictactoc/MyAdapter.kt
package com.raeed.tictactoc

import android.content.Context
import com.raeed.tictactoc.HistoryArray
import android.widget.BaseAdapter
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import com.raeed.tictactoc.R
import android.widget.TextView
import java.util.ArrayList

class MyAdapter(
      private val context: Context,
      resorsec: Int,
      historyArrays: ArrayList<HistoryArray>
) : BaseAdapter() {
      var information: HistoryArray? = null
      private var historyArrays = ArrayList<HistoryArray>()
      private val resorsec: Int

      init {
            this.historyArrays = historyArrays
            this.resorsec = resorsec
      }

      override fun getCount(): Int {
            return historyArrays.size
      }

      override fun getItem(position: Int): Any {
            information = historyArrays[position]
            return information as HistoryArray
      }

      override fun getItemId(position: Int): Long {
            return 0
      }

      override fun getView(position: Int, convertView: View, parent: ViewGroup): View {
            var view: View
            view = convertView
            if (view == null) {
                  view = LayoutInflater.from(context).inflate(R.layout.listviewlayout, null, false)
            }
            val player_A_Image = view.findViewById<ImageView>(R.id.Aimage)
            val player_B_Image = view.findViewById<ImageView>(R.id.Bimage)
            val aVsb = view.findViewById<TextView>(R.id.AvsB)
            val serialnumber = view.findViewById<TextView>(R.id.number)
            val player_A_ruselt = view.findViewById<TextView>(R.id.Atext)
            val player_B_ruselt = view.findViewById<TextView>(R.id.Btext)
            player_A_Image.setImageResource(R.drawable.test)
            player_B_Image.setImageResource(R.drawable.test)
            val player_A = information!!.player_A // Todo حل مشكلة null object refernce
            val player_B = information!!.player_B
            aVsb.text = "$player_A Vs $player_B"
            serialnumber.text = position.toString() + ""
            val id = information!!.winer_Id
            if (id == 0) {
                  player_A_ruselt.text = "winer"
                  player_B_ruselt.text = "Looser"
            } else {
                  player_A_ruselt.text = "Looser"
                  player_B_ruselt.text = "winer"
            }
            return view
      }
}